projects/ngrx.io/content/guide/signals/signal-store/testing.md

```diff
@@ -0,0 +1,687 @@
+<!-- TOC -->
+  * [Introduction](#introduction)
+    * [On Testing in general](#on-testing-in-general)
+    * [What to test](#what-to-test)
+    * [TestBed or not?](#testbed-or-not)
+  * [Testing the Signal Store](#testing-the-signal-store)
+    * [Globally provided](#globally-provided)
+    * [Locally provided](#locally-provided)
+    * [`withComputed`](#withcomputed)
+    * [`withMethods`, DI, and asynchronous tasks](#withmethods-di-and-asynchronous-tasks)
+    * [`rxMethod`](#rxmethod)
+      * [with Observables](#with-observables)
+      * [with Signals](#with-signals)
+    * [Custom extensions](#custom-extensions)
+  * [Mocking the Signal Store](#mocking-the-signal-store)
+    * [Native Mocking](#native-mocking)
+    * [ng-mocks](#ng-mocks)
+    * [Partial Mocking via Spies](#partial-mocking-via-spies)
+<!-- TOC -->
+
+This is still a draft. Markdown is used for better readability. The rendered version of this guide is available at https://github.com/rainerhahnekamp/ngrx/blob/docs/signals/testing/projects/ngrx.io/content/guide/signals/signal-store/testing.md.
+
+Once the content is finalized, we will replace markdown with `<code>` tags.
+
+The examples used in this guide are available at https://github.com/rainerhahnekamp/ngrx-signal-store-testing
+
+## Introduction
+
+### On Testing in General
+
+A Signal Store is a simple Angular Service, so the same testing techniques you use for services apply to Signal Stores as well. This guide focuses on providing examples for common testing scenarios.
+
+A challenging part of testing is knowing how to handle asynchronous tasks and mocking. The examples use Jest (v29.5), but the same principles apply to other testing frameworks.
+
+There are two main scenarios for testing:
+
+1. Testing the `signalStore` itself,
+2. Testing a component or service that uses the Signal Store.
+
+In the first scenario, you’ll mock the dependencies of the Signal Store, while in the second scenario, you’ll mock the Signal Store itself.
```

Comment:
There is a third scenario where we test the service by:
1. creating a testing component
2. injecting the service
3. mocking only the service dependencies (not the signal store)

In this way, we can make assertions about what user behavior the service provides without testing the implementation details of the service.

Scenario one above tests pure implementation details and tells us very little about how the service will be used alongside user behavior. Scenario two above tests user behavior, but depending on how broad the scope of the service is may not result in 100% coverage of the branches/functions/etc of the service in that single component.

I think there's a good opportunity to present an opinion about how services can be tested in general (via this third scenario).

I think Tim is hinting at this in his [comment about integration testing](https://github.com/ngrx/platform/pull/4461/files#r1704466865) but I don't want to put words in his mouth

Reply:
Yes, I thought it is not necessary, but you two have convinced me. I will add it.